### PR TITLE
Add initial EMR cluster Terraform module

### PR DESCRIPTION
**Pull request by hectcastro**
Use Terraform configuration to define all of the AWS resources needed to
support an Amazon EMR cluster. This includes:

- IAM roles for cluster instances
- Skeleton security groups for the cluster instances
- The EMR cluster itself

Resolves https://github.com/azavea/operations/issues/128

---

**Testing**

See testing instructions for https://github.com/azavea/raster-foundry-deployment/pull/93.

### CODE REVIEW

**Comment by hectcastro** [comment]
@lossyrob I believe that this PR addresses what's necessary for https://github.com/azavea/osmesa/issues/3 to happen via this module. There is a mechanism to provide bootstrap scripts with arguments, and we can override the configuration JSON via `configurations`. Also, spot pricing can be handled via the `instances` object.

If its helpful, https://github.com/azavea/raster-foundry-deployment/pull/93 shows how we might make use of this module.

Please comment here if you think something is missing or needs to be tweaked.

**Comment by lossyrob** [comment]
@hectcastro thanks, I'll look to that PR for implementation.

Do you have any guidance on separating out various components of the infrastructure, to be brought up and down independently? My current approach is to separate out terraform files, e.g.

```
deployment/terraform/analytics/*.tf
deployment/terraform/ingest/*.tf
deployment/terraform/update/*.tf
... etc
```

and then have separate `scripts/analytics.sh` etc to deploy specific aspects, as opposed to a single `infra.sh`.

Also, I want to allow the option of having local `tfvars`, so that you don't necessarily have to upload and download them during development. Currently the approach is to have this in the script:

```sh
    if [[ -n "${OSMESA_SETTINGS_BUCKET}" ]]; then
        aws s3 cp "s3://${OSMESA_SETTINGS_BUCKET}/terraform/osmesa/terraform.tfvars" \
            "${TERRAFORM_DIR}/${OSMESA_SETTINGS_BUCKET}.tfvars"
        TERRAFORM_SETTINGS=${OSMESA_SETTINGS_BUCKET}.tfvars
    else
        TERRAFORM_SETTINGS=${PWD}/deployment/terraform/terraform.tvars
    fi
```

If all this sounds sane, I'll move forward with it, unless you have suggestions on a better approach. Thanks

**Comment by lossyrob** [comment]
Looks like rasterfoundry deployment uses a singe `infra` script, and uses `--arg` to differentiate components. I'll mimic this.

**Comment by lossyrob** [comment]
Moving my comments to OSMesa, where I thought they were originally

File: README.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+```hcl
+data "template_file" "emr_configurations" {
+  template = "${file("configurations/default.json")}"
+}
+
+module "emr" {
+  source = "github.com/azavea/terraform-aws-emr-cluster?ref=0.1.0"
+
+  name          = "DatarpocCluster"
+  vpc_id        = "vpc-20f74844"
```
Comment (tnation14) [inline README.md:16]: Not sure if this is an actual VPC id but I figured I'd draw attention to it anyway.

**Review by tnation14 — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (hectcastro) [reply README.md:16]: I just mashed the keyboard.

**Pull request merged**